HtmRouter.js

```diff
@@ -1,27 +1,54 @@
+/*global useState */
+// Note that this means you have to push useState into global scope
+// in your code. See https://github.com/ruffin--/preact-router-for-htm for more.
 (function () {
+    function isRegExp(obj) {
+        return Object.prototype.toString.call(obj) === '[object RegExp]';
+    }
+
+    function endsWithViaRegExp(str, re) {
+        var reString = re.source;
+        re = reString.endsWith('$') ? re : new RegExp(reString + '$', re.flags);
+        return re.test(str);
+    }
+
+    function endsWithDeluxe(str, stringOrRegExp) {
+        return isRegExp(stringOrRegExp)
+            ? endsWithViaRegExp(str, stringOrRegExp)
+            : str.endsWith(stringOrRegExp);
+    }
+
     window.HtmRouter = function (props) {
         // =============================================
         // jive stolen from
         // https://github.com/preactjs/preact-router
         // =============================================
         function prevent(e) {
-            if (e.stopImmediatePropagation) e.stopImmediatePropagation();
-            if (e.stopPropagation) e.stopPropagation();
+            if (e.stopImmediatePropagation) {
```

Comment:
Looks like the prettier max length is a little different than what was originally used, but this probably looks better.